File: 7-kyu/duck-shoot---easy-version/c++/solution.cpp
#include <cmath>
#include <string>

std::string duckShoot(int ammo, double aim, std::string ducks)
{
    int ducksShot = std::floor(static_cast<double>(ammo) * aim);
    for (auto &c : ducks) {
        if (c == '2' && ducksShot > 0) {
            c = 'X';
            ducksShot--;
        }
    }
    return ducks;
}

File: 7-kyu/duck-shoot---easy-version/python/solution.py
from math import floor

def duck_shoot(ammo, aim, ducks):
    ducks_to_shoot = floor(ammo * aim)
    ret = [ch for ch in ducks]
    for i, target in enumerate(ducks):
        if target == '2' and ducks_to_shoot > 0:
            ret[i] = 'X'
            ducks_to_shoot -= 1
    return ''.join(ret)
